src/components/PreferencesBodyKubernetes.vue

```diff
@@ -0,0 +1,157 @@
+<script lang="ts">
+import Vue from 'vue';
+import { ipcRenderer } from 'electron';
+import Checkbox from '@/components/form/Checkbox.vue';
+import { Settings } from '@/config/settings';
+import { VersionEntry } from '@/k8s-engine/k8s';
+import InputLegend from '@/components/form/InputLegend.vue';
+
+export default Vue.extend({
+  name:       'preferences-body-kubernetes',
+  components: { Checkbox, InputLegend },
+  data() {
+    return {
+      enableKubernetes: true,
+      enableTraefik:    true,
+      kubernetesPort:   6443,
+      settings:         { kubernetes: {} } as Settings,
+      versions:         [] as VersionEntry[],
+    };
+  },
+  computed: {
+    defaultVersion(): VersionEntry {
+      const version = this.recommendedVersions.find(v => (v.channels ?? []).includes('stable'));
+
+      return version ?? (this.recommendedVersions ?? this.nonRecommendedVersions)[0];
+    },
+    /** Versions that are the tip of a channel */
+    recommendedVersions(): VersionEntry[] {
+      return this.versions.filter(v => !!v.channels);
+    },
+    /** Versions that are not supported by a channel. */
+    nonRecommendedVersions(): VersionEntry[] {
+      return this.versions.filter(v => !v.channels);
+    },
+    isKubernetesDisabled(): boolean {
+      return !this.enableKubernetes;
+    }
+  },
+  beforeMount() {
+    ipcRenderer.on('k8s-versions', (event, versions) => {
+      this.versions = versions;
+      this.settings.kubernetes.version = this.defaultVersion.version.version;
+    });
+
+    ipcRenderer.send('k8s-versions');
+  },
+  methods: {
+    onChange() {
+      this.$emit('change:kubernetes-version');
+    },
+    /**
+     * Get the display name of a given version.
+     * @param version The version to format.
+     */
+    versionName(version: VersionEntry) {
+      const names = (version.channels ?? []).filter(ch => !/^v?\d+/.test(ch));
+
+      if (names.length > 0) {
+        return `v${ version.version.version } (${ names.join(', ') })`;
+      }
+
+      return `v${ version.version.version }`;
+    },
+  }
+});
+</script>
+
+<template>
+  <div class="preferences-body">
+    <input-legend>
+      <template #legend>
+        <span>Kubernetes</span>
+      </template>
+      <checkbox
+        v-model="enableKubernetes"
+        label="Enable Kubernetes"
+      />
+    </input-legend>
+    <input-legend class="w-half">
```

Comment:
Presumably `w-half` is because the thing is too wide otherwise? We don't have to care about it as much right now (because there are few things here), but we may want to consider moving things around later.

Reply:
Yeah, it doesn't make sense for these inputs to be full width. Neither the kubernetes version or the port really require that much space..